common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -22,5 +22,79 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.ui.embeddedwebview.challengehandlers;
 
-final class ClientCertAuthChallengeHandler {
-}
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.os.Build;
+import android.security.KeyChain;
+import android.security.KeyChainAliasCallback;
+import android.security.KeyChainException;
+import android.support.annotation.NonNull;
+import android.webkit.ClientCertRequest;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+public final class ClientCertAuthChallengeHandler implements IChallengeHandler {
+    private static final String TAG = ClientCertAuthChallengeHandler.class.getSimpleName();
+    private ClientCertRequest mClientCertRequest;
+    private Activity mActivity;
+
+    public ClientCertAuthChallengeHandler(@NonNull final ClientCertRequest request,

```

Comment:
The `ClientCertRequest` is basically "the challenge", right? Unlike the other `ChallengeHandler` classes, this one appears to accept the Challenge and not construct it (which I like).

Consider the suggestion [from here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/158#discussion_r200779395) --  you could declare this class using `IChallengeHandler<ClientCertRequest, Void>` and users of this class could conceivably swap out its implementation if they wanted to